oneibl/webclient.py

```diff
@@ -53,6 +54,74 @@ def __iter__(self):
             yield self.__getitem__(i)
 
 
+def sdsc_globus_path_from_dataset(dset):
+    """
+    :param dset: dset dictionary or list of dictionaries from ALyx rest endpoint
+    Returns SDSC globus file path from a dset record or a list of dsets records from REST
+    """
+    return _path_from_dataset(dset, root_path=None, repository=None, uuid=True)
+
+
+def globus_path_from_dataset(dset, repository=None, uuid=False):
+    """
+    Returns local one file path from a dset record or a list of dsets records from REST
+    :param dset: dset dictionary or list of dictionaries from ALyx rest endpoint
+    :param repository: (optional) repository name of the file record (if None, will take
+     the first filerecord with an URL)
+    """
+    return _path_from_dataset(dset, root_path=None, repository=repository, uuid=uuid)
+
+
+def one_path_from_dataset(dset, one_cache):
+    """
+    Returns local one file path from a dset record or a list of dsets records from REST
+    :param dset: dset dictionary or list of dictionaries from ALyx rest endpoint
+    :param one_cache: the one cache directory
+    """
+    return _path_from_dataset(dset, root_path=one_cache, uuid=False)
+
+
+def sdsc_path_from_dataset(dset, root_path=SDSC_ROOT_PATH):
+    """
+    Returns sdsc file path from a dset record or a list of dsets records from REST
+    :param dset: dset dictionary or list of dictionaries from ALyx rest endpoint
+    :param root_path: (optional) the prefix path such as one download directory or sdsc root
+    """
+    return _path_from_dataset(dset, root_path=root_path, uuid=True)
+
+
+def _path_from_dataset(dset, root_path=None, repository=None, uuid=False):
+    """
+    returns the local file path from a dset record from a REST query
+    :param dset: dset dictionary or list of dictionaries from ALyx rest endpoint
+    :param root_path: (optional) the prefix path such as one download directory or sdsc root
+    :param repository:
+    :param uuid: (optional bool) if True, will add UUID before the file extension
+    :return: Path or list of Path
+    """
+    if isinstance(dset, list):
+        return [_path_from_dataset(d) for d in dset]
+    if repository:
+        fr = next((fr for fr in dset['file_records'] if fr['data_repository'] == repository))
+    else:
+        fr = next((fr for fr in dset['file_records'] if fr['data_url']))
+    uuid = dset['url'][-36:] if uuid else None
+    return _path_from_filerecord(fr, root_path=root_path, uuid=uuid)
+
+
+def _path_from_filerecord(fr, root_path=SDSC_ROOT_PATH, uuid=None):
+    import alf.io
```

Comment:
What's the reason for importing within the function?

Reply:
those nasty brainbox circular imports